### 문제1
   - a="Hello Python" 문자열을 Slicing과 연결 연산자 (+)를 사용하여 "Python Hello"로 변경하시오.
     - [주의] a 변수 자체에 "Python Hello" 문자열이 저장되어야 함

In [1]:
a="Hello Python"
print(a)
a=a[6:]+' '+a[:5]
print(a)

Hello Python
Python Hello


슬라이싱을 이용하여 a="Hello Python"를 바꾸기 위해 P의 시작 부분 [6:]부터 시작하여 
마지막까지를 집어넣고, 앞부분의 마지막인 o는 a[4]이므로 a[:5]로 하여 "Hello를 불러낸다.

### 문제2
  - b="Hello Python World" 문자열을 Slicing과 연결 연산자 (+)를 사용하여 "World Python Hello"로 변경하시오.
     - [주의] b 변수 자체에 "Python Hello" 문자열이 저장되어야 함

In [2]:
b="Hello Python World"
print(b)
b=b[13:]+' '+b[6:13]+b[:6]
print(b)

Hello Python World
World Python Hello 


문제1과 비슷하게 슬라이싱을 이용하여 W의 시작부분 b[13]을 이용, P의 시작부분 b[6]과 마지막 부분 n은 b[12]을 이용,마지막 o는 b[5]이므로 이용하여 값을 구해냈다.

### 문제3
- c="Hello"를 "olleH"로 변경하시오
     - [주의] c 변수 자체에 "olleH" 문자열이 저장되어야 함

In [3]:
c="Hello"
print(c)
c=c[::-1]
print(c)

Hello
olleH


배열을 반대로 바꾸는 것이므로, 뒤에서 부터 나오도록c[::-1]로 하여 하면 반대로 실행이 되게된다.

### 문제4
  - s="python"에 대해 다음 문제를 풀어보시오.
    1. s[0], s[0][0], s[0][0][0]은 각각 어떤 값이 나오는지 확인하고 그 이유를 나름대로 설명해 보시오.
    2. s[-100], s[100]은 값이 나오는지 에러가 나오는지 확인하고 그 결과에 대한 이유를 나름대로 설명해 보시오.
    3. s[-100, 100]은 값이 나오는지 에러가 나오는지 확인하고 그 결과에 대한 이유를 나름대로 설명해 보시오.
    4. s[1:-1]의 결과를 확인하고 그 결과에 대한 이유를 정확하게 설명하시오.
    5. s[3:-3]의 결과를 확인하고 그 결과에 대한 이유를 정확하게 설명하시오.

  1. s[0]와 s[0][0], s[0][0][0]는 모두 값이 같다. 1차원 배열은 s[0]일 때, p인것이 확실하고 s[0][0]도 또한 y축이 있더라도 값이 0이므로 y[0]인 s는 s[0]과 값이 같고, s[0][0][0]또한 같다.
  2. c[100]과 c[-100]은 값이 나오지 않는다. 그 이유는 s 문자열은 6개로 이루어져있다. 그러므로 6을 초과한 100과 -100은 실행되지 않는다.
  3. s[-100,100] 또한 1.과 2.의 이유로 배열의 값이 초과하였고, 2차원이 0이 아닌 값으로 초과 하였다.
  4. s[1:-1]는 값이 ytho가 나온다. 그 이유는 첫값을 1로 시작하여 s[1]는 'y'이고 [:-1]은 끝을 나타내므로 -1은 또 뒤에서 부터 새는것 이므로 s[-1] n에 해당되므로 ytho가 된다.
  5. s[3:-3]은 값이 없다. 왜냐하면 s[3]은 h이고, s[-3]은 또 h이다. 하지만 슬라이싱은 마지막 h는 포함되지 않기 때문에 값이 출력 되지않는다.

### 문제5
  - for문을 활용하여 1부터 100사이의 홀수를 출력하시오.

In [4]:
result = 0
d=range(100)
for x in d:
    if x%2==1:
        result += x
print(result)

2500


for을 이용하여 x가 1~100까지의 숫자를 얻도록 range(100)을 받고, 받은 값들을 x%2==1을 하면 홀수 값이 나오므로 이값들을 다 더하면 1~100까지의 홀수인 2500이 출력된다.

### 문제6
  - while문을 활용하여 1부터 100사이의 짝수의 합을 계산하여 출력하시오.

In [5]:
count=1
sum=0
while count<=100:
    count+=1
    if count%2==0:
        sum += count
print(sum)

2550


문제5와 비슷하여 while문을 이용하여 count를 이용해 1~100까지의 수를 실행 실행하여 count가 1부터 값이 올라가면서 count%2==0을 실행하여 짝수인 값을 계속 더하여 출력하면 2550이 나온다.

### 문제7
  - 사용자로 부터 임의의 정수를 입력받고, 해당 숫자를 역순으로 출력하는 프로그램을 작성하시오.

In [6]:
a=input("정수를 입력하세요:")
a=a[::-1]
print("숫자 역순은",a,"입니다.")

정수를 입력하세요:3125
숫자 역순은 5213 입니다.


역순으로 실행하기 위해서는 문제 3과 똑같은 방식으로 풀면된다. 바로 [::-1]을 이용하면 역순으로 실행되기 때문에 input으로 값을 받으면 [::-1]를 이용해 역순값을 받는다.

### 문제8
  - 사용자로 부터 정수를 입력받아서 1부터 그 사이에 존재하는 소수 (Prime number)를 출력하는 파이썬 프로그램을 작성하시오.

In [7]:
count=1
enter=int(input("정수를 입력하세요: "))
while(count<=enter):
    if enter%count==0:
        print(count)
    count+=1

정수를 입력하세요: 6
1
2
3
6


먼저 소수인 수를 나타내기위한 정수를 받기위해서 int(input())을 이용해 정수값을 받고, 그 정수값을 while문 형태에서 count를 받아 값을 돌리면서 count+=1을 하면서 
enter%count가 0이면 소수이므로 그 값을 출력한다.

### [ACM-ICPC 문제 2439번]
  - 첫째 줄에는 별 1개, 둘째 줄에는 별 2개, N번째 줄에는 별 N개를 찍는 문제

In [12]:
count=1
n=int(input())
while n>=count:
    print (" "*(n-count)+"*"*count)
    count+=1

5
    *
   **
  ***
 ****
*****


먼저 int(input())으로 값을 받고, 그 값을 별 1개씩 찍기위해서는 " "는 n-count와 같고, " "는 count와 개수가 맞게 움직이므로 count+=1하면서 실행하면 결과값이 출력된다.
<img src="files/1.PNG">

### [ACM-ICPC 문제 1924번]
  - 오늘은 2007년 1월 1일 월요일이다. 그렇다면 2007년 x월 y일은 무슨 요일일까? 이를 알아내는 프로그램을 작성하시오.

In [27]:
count=0
add=0
result=0
x=int(input("월 입력: "))
y=int(input("일 입력: "))
while(x>count):
    if count==1:
        add+=31
    elif count==3:
        add+=31
    elif count==5:
        add+=31
    elif count==7:
        add+=31
    elif count==10:
        add+=31
    elif count==12:
        add+=31         
    elif count==4:
        add+=30
    elif count==6:
        add+=30
    elif count==8:
        add+=30
    elif count==9:
        add+=30
    elif count==11:
        add+=30
    elif count==2:
        add+=28
    else:
        add+=0
    result=add+y    
    count+=1   
result=result%7
if result==1:
    print("MON")
elif result==2:
    print("TUE")
elif result==3:
    print("WED")
elif result==4:
    print("THU")
elif result==5:
    print("FRI")
elif result==6:
    print("SAT")
else:
    print("SUN")


월 입력: 1
일 입력: 1
MON


먼저 월이 각자 마지막 일이 다르므로 모든 그 값들에 대해 if문을 이용해 add에 더해준다.
만약 3이라면 count를 이용해 count==1과2 가 실행되어 31+28이 add에 더해진다. 그 후 일에서 입력 받은 y값도 더해준다. 그리고 그 값을 또 %7로 하여 1은 MON 2는 TUE 이런식으로 출력되도록 if문을 설정한다.
<img src="files/2.PNG">
jupyter notebook에서는 실행되나 사이트에서는 런타임 에러가 뜬다.

### [ACM-ICPC 문제 11720번]
  - N개의 숫자가 공백 없이 쓰여있다. 이 숫자를 모두 합해서 출력하는 프로그램을 작성하시오.

In [5]:
count=0
result=0
x=int(input())
y=input()
while(count<x):
    result+=int(y[count])
    count+=1
print(result)

5
54321
15


먼저 x는 int(input())를 이용해 정수로 받고, y는 input()를 이용해 값을 받는다. 그 후 while를 위해 count를 +1씩 더해주면서 x와 값이 같아질 때까지 돌린다. 이 때, 스트링인 y를 count를 이용해 0부터 x까지의 스트링을 하나씩 빼오면서 int형으로 바꾸면서 더해준다. 이렇게해서 다 더해주면 올바른 결과값이 출력된다.
<img src="files/3.png">